File: examples/callback_example.cxx
#include "egs.h"
#include "callback_plugin.hxx"

int main(void) {
  Context ctx;
  GLFWContext glfw_ctx = GLFWContext(ctx);
  auto display_list = std::make_shared<DisplayList>();

  const char *text = "Hello EGS!";
  display_list->add(std::make_shared<CallbackPlugin::Callback>([&text] (GLContext& gl_ctx) {
    (void)gl_ctx;
    std::cout << text << std::endl;
  }));

  while (glfw_ctx.update(display_list));

  return 0;
}

File: examples/callback_example.py

from __future__ import print_function

import egs
from egs.plugins.callback_plugin import Callback

context = egs.Context()
glfw_context = egs.GLIPGLFWContext(context)
display_list = egs.DisplayList()

display_list.add(Callback(lambda gl_ctx: print("Hello EGS!")))

while glfw_context.update(display_list):
    pass
